hadoop-tools/hadoop-azure/src/main/java/org/apache/hadoop/fs/azurebfs/services/AbfsInputStreamStatisticsImpl.java

```diff
@@ -90,9 +107,7 @@ public void seek(long seekTo, long currentPos) {
    */
   @Override
   public void bytesRead(long bytes) {
-    if (bytes > 0) {
-      bytesRead += bytes;
-    }
+    ioStatisticsStore.incrementCounter(STREAM_READ_BYTES, bytes);
```

Comment:
One thing to consider here is the cost of the map lookup on every IOP. You can ask the IOStatisticsStore for a reference to the atomic counter, and use that direct. I'm doing that for the output stream, reviewing this patch makes me realise I should be doing it for read as well. at least the read byte counters which are incremented on every read.

      bytesUploaded = store.getCounterReference(
          STREAM_WRITE_TOTAL_DATA.getSymbol());
      bytesWritten = store.getCounterReference(
          StreamStatisticNames.STREAM_WRITE_BYTES);

Reply:
Thanks for the tip. I've added the frequently updated counters as counter ref.